src/galaxy/SectorGenerator.h

```diff
@@ -13,19 +14,26 @@
 class SectorCustomSystemsGenerator : public SectorGeneratorStage {
 public:
 	SectorCustomSystemsGenerator(int customOnlyRadius) :
-		m_customOnlyRadius(customOnlyRadius) {}
+		m_customOnlyRadius(customOnlyRadius) {
+		SectorCustomSystemsGenerator::galaxyConfig = new GalaxyConfig();
+	}
 	virtual bool Apply(Random &rng, RefCountedPtr<Galaxy> galaxy, RefCountedPtr<Sector> sector, GalaxyGenerator::SectorConfig *config);
 
 private:
 	int m_customOnlyRadius;
+	static GalaxyConfig *galaxyConfig;
 };
 
 class SectorRandomSystemsGenerator : public SectorGeneratorStage {
 public:
+	SectorRandomSystemsGenerator(){
+		SectorRandomSystemsGenerator::galaxyConfig = new GalaxyConfig();
+	}
 	virtual bool Apply(Random &rng, RefCountedPtr<Galaxy> galaxy, RefCountedPtr<Sector> sector, GalaxyGenerator::SectorConfig *config);
 
 private:
 	const std::string GenName(RefCountedPtr<Galaxy> galaxy, const Sector &sec, Sector::System &sys, int si, Random &rand);
+	static GalaxyConfig *galaxyConfig;
```

Comment:
done?
I just removed the static part, and have it as a normal member variable (not pointer). Feeling quite unsure about what I'm doing here, but it compiles & runs & works.
I used to know C++....

Reply:
Static member variables are "one copy shared by all instances of the class" - each time a new generator object is constructed, it was overwriting the old generator object at the single shared location in memory. This technically isn't a problem until you throw multithreading into the mix, at which point it can become a huge footgun if you're not careful (e.g. one thread writes, the other thread reads).

While that's not an extant concern here, I'd still discourage the use of a static member variable to stick the config in - the new implementation is better.